harness/determined/profiler.py

```diff
@@ -0,0 +1,524 @@
+        self.start_sent = net.bytes_sent
+        self.start_recv = net.bytes_recv
+
+    def measure(self, batch_idx):
+        net = psutil.net_io_counters()
+        end_time = time.time()
+
+        sent_bytes_delta = net.bytes_sent - self.start_sent
+        recv_bytes_delta = net.bytes_recv - self.start_recv
+
+        time_delta = end_time - self.start_time
+
+        self.start_time = end_time
+        self.start_sent = net.bytes_sent
+        self.start_recv = net.bytes_recv
+
+        sent_throughput_bytes_per_second = sent_bytes_delta / time_delta
+        recv_throughput_bytes_per_second = recv_bytes_delta / time_delta
+
+        sent_throughput_gigabits_per_second = sent_throughput_bytes_per_second * 8 * GIGA
+        recv_throughput_gigabits_per_second = recv_throughput_bytes_per_second * 8 * GIGA
+
+        timestamp = datetime.datetime.fromtimestamp(end_time)
+        return Measurement(timestamp, batch_idx, sent_throughput_gigabits_per_second), \
+               Measurement(timestamp, batch_idx, recv_throughput_gigabits_per_second)
+
+
+class DiskReadWriteRateCollector:
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.start_time = time.time()
+        disk = psutil.disk_io_counters()
+
+        self.start_read_bytes = disk.read_bytes
+        self.start_write_bytes = disk.write_bytes
+
+        self.start_read_count = disk.read_count
+        self.start_write_count = disk.write_count
+
+    def measure(self, batch_idx):
+        disk = psutil.disk_io_counters()
+        end_time = time.time()
+
+        read_bytes_delta = disk.read_bytes - self.start_read_bytes
+        write_bytes_delta = disk.write_bytes - self.start_write_bytes
+
+        read_count_delta = disk.read_count - self.start_read_count
+        write_count_delta = disk.write_count - self.start_write_count
+
+        time_delta = end_time - self.start_time
+
+        self.start_time = end_time
+        self.start_read_bytes = disk.read_bytes
+        self.start_write_bytes = disk.write_bytes
+        self.start_read_count = disk.read_count
+        self.start_write_count = disk.write_count
+
+        read_throughput_bytes_per_second = read_bytes_delta / time_delta
+        write_throughput_bytes_per_second = write_bytes_delta / time_delta
+
+        read_throughput_count_per_second = read_count_delta / time_delta
+        write_throughput_count_per_second = write_count_delta / time_delta
+
+        timestamp = datetime.datetime.fromtimestamp(end_time)
+        read_throughput = Measurement(timestamp, batch_idx, read_throughput_bytes_per_second)
+        write_throughput = Measurement(timestamp, batch_idx, write_throughput_bytes_per_second)
+        iops = Measurement(timestamp, batch_idx, read_throughput_count_per_second + write_throughput_count_per_second)
+
+        return read_throughput, write_throughput, iops
+
+
+
+class GpuUtilCollector:
+    def __init__(self):
+        pynvml.nvmlInit()
```

Comment:
I'm adding this to environments, we can do something like
```
try:
    import pynvml
    pynvml.nvmlInit()
except ModuleNotFoundError:
    pass
except pynvml.NVMLError_LibraryNotFound:
    pass
```
here to check whether we should enable this collector or not.